### ISSUE TITLE
Given empty pathvariable value and get 500 internal server error

### ISSUE THREAD

**Issue by LinhaoJiaTomo** [springdoc/springdoc-openapi#2203]
**Describe the bug**
We have an api like
```
@GetMapping("/xxx/{id}")
fun retrieveXxxById(@PathVariable id: String)
```

And when we call this api using `/xxx` and not give the PathVariable id, we'll get 500 error, this issue happens when we update our org.springdoc:springdoc-openapi-kotlin from 1.6.15 to 1.7.0, it used to throw 404 from our logic.
Our configuration of springdoc is
```
springdoc:
  webjars:
    prefix: ''
  swagger-ui:
    disable-swagger-default-url: true
    operationsSorter: method
    tagsSorter: alpha
  api-docs:
    enabled: true
```

And the exception detail is
```
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(Unknown Source)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Original Stack Trace:
		at java.lang.String.substring(Unknown Source)
		at org.springdoc.ui.AbstractSwaggerResourceResolver.path(AbstractSwaggerResourceResolver.java:71)
		at org.springdoc.ui.AbstractSwaggerResourceResolver.findWebJarResourcePath(AbstractSwaggerResourceResolver.java:42)
		at org.springdoc.webflux.ui.SwaggerResourceResolver.resolveResource(SwaggerResourceResolver.java:35)
		at org.springframework.web.reactive.resource.DefaultResourceResolverChain.resolveResource(DefaultResourceResolverChain.java:75)
		... 148 frames truncated

```

**To Reproduce**

- What version of spring-boot you are using?
`2.7.10`
- What modules and versions of springdoc-openapi are you using?
`org.springdoc:springdoc-openapi-webflux-ui:1.7.0`, `org.springdoc:springdoc-openapi-kotlin:1.7.0`

**Expected behavior**
Not return 500

**Comment by sanketmeghani** [comment]
I am also getting similar error while accessing `/swagger-ui/`.

**Comment by Martin01248** [comment]
I am also having this issue that `/swagger-ui` leads to `StringIndexOutOfBoundsException: String index out of range: -1` resulting in 500 server error.

Class `AbstractSwaggerResourceResolver`
with method
```
	private String path(String webjar, String path) {
		if (path.startsWith(webjar)) {
			path = path.substring(webjar.length() + 1);
		}
		return path;
	}
```
where parameters are
`webjar="swagger-ui"`
`path="swagger-ui"`
resulting in
`StringIndexOutOfBoundsException: String index out of range: -1`

**Comment by uc4w6c** [comment]
i think this will fix it. #2244

**Comment by bnasslahsen** [comment]
Fixed by #2235
